File: hto/53/hto_53.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int lo = 0, hi = nums.size();
        while (lo < hi) {
            int mid = (lo + hi) >> 1;
            if (target <= nums[mid])
                hi = mid;
            else 
                lo = mid + 1;
        }
        if (lo == nums.size() || nums[lo] != target) return 0;
        int start = lo;

        hi = nums.size();
        while (lo < hi) {
            int mid = (lo + hi) >> 1;
            if (target >= nums[mid])
                lo = mid + 1;
            else
                hi = mid;
        }
        return hi - start;
    }
};

File: hto/53/hto_53.py
from typing import List, Dict
from queue import Queue
from collections import deque, Counter
import math
from heapq import heappush, heappushpop


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def binarySearch(nums: List[int], target: int):
            lo, hi = 0, len(nums)
            while lo < hi:
                mid = (lo + hi) >> 1
                if target <= nums[mid]:
                    hi = mid
                else:
                    lo = mid + 1
            return lo

        start = binarySearch(nums, target)
        end = binarySearch(nums, target + 1)
        if start == len(nums) or nums[start] != target:
            return 0
        else:
            return end - start


if __name__ == "__main__":
    arr = [5,7,7,8,8,10]
    s = Solution()
    print(s.search(arr, 8))
